script/pr-api.ts

```diff
@@ -0,0 +1,117 @@
+import * as HTTPS from 'https'
+
+export interface IAPIPR {
+  readonly title: string
+  readonly body: string
+  readonly headRefName: string
+  readonly permalink: string
+}
+
+type GraphQLResponse = {
+  readonly data: {
+    readonly repository: {
+      readonly pullRequest: IAPIPR
+    }
+  }
+}
+
+function gitHubRequest(
+  options: HTTPS.RequestOptions,
+  body: Record<string, any>
+): Promise<Record<string, any> | null> {
+  const opts: HTTPS.RequestOptions = {
+    host: 'api.github.com',
+    protocol: 'https:',
+    method: 'POST',
+    headers: {
+      Authorization: `bearer ${process.env.GITHUB_ACCESS_TOKEN}`,
+      'User-Agent': 'what-the-changelog',
+    },
+    ...options,
+  }
+
+  return new Promise((resolve, reject) => {
+    const request = HTTPS.request(opts, response => {
```

Comment:
Just curious why you chose HTTP.request as opposed to fetch? I haven't written a vanilla request in a while, but was thinking fetch was simpler to use.

Reply:
I didn't either! Then I tried to write the same using `fetch` and got a `fetch is not available` error 😂  Then I understood why it used `HTTPS.request`… 👀